### PR TITLE
XML export fix tests. Add not nil room for scheduled events.

### PR DESCRIPTION
**Pull request by differentreality**

### CODE REVIEW

**Comment by differentreality** [comment]
@hennevogel  I have trouble defining this in a factory. Due to program being created automatically on conference creation?
https://github.com/differentreality/osem/blob/15076b11f14701d697a5e691d32332348da6b7a9/spec/controllers/conference_controller_spec.rb#L32

**Comment by hennevogel** [comment]
@differentreality do it in an `after(:create)` block?

File: app/models/program.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def confirmed
     end
 
     def scheduled
-      where.not(start_time: nil)
+      where.not(start_time: nil).where.not(room: nil)
```
Comment (hennevogel) [inline app/models/program.rb]: Hm. How can this happen?

Comment (differentreality) [inline app/models/program.rb]: I guess the same way we currently have the tours scheduled with time, but without room. (which is, I think, what made XML export fail)

Comment (hennevogel) [inline app/models/program.rb]: okay

**Pull request merged**